WNPRC_Purchasing/src/client/components/LineItemRow.tsx

```diff
@@ -47,52 +52,78 @@ export const LineItemRow: FC<LineItemProps> = memo(props => {
-                        onClick={onDeleteRow}
-                    >
-                        <FontAwesomeIcon className="fa-faTimesCircle" icon={faTimesCircle} />
-                    </span>
-                </Col>
-            </Row>
+            {
+                <Row className="line-item-row" key={'line-item-row-' + rowIndex}>
+                    <Col xs={4}>
+                        <DescriptionInput
+                            value={model.item}
+                            hasError={model.errors?.find(field => field.fieldName === 'item')}
+                            onChange={onValueChange}
+                            isReadOnly={!isAdmin && canUpdate}
+                        />
+                    </Col>
+                    <Col xs={1}>
+                        <UnitInput
+                            value={model.itemUnit}
+                            hasError={model.errors?.find(field => field.fieldName === 'itemUnit')}
+                            onChange={onValueChange}
+                            isReadOnly={!isAdmin && canUpdate}
+                        />
+                    </Col>
+                    { (isAdmin || !canUpdate) &&
+                        <Col xs={1}>
+                            <UnitCostInput
+                                    value={model.unitCost}
+                                    hasError={model.errors?.find(field => field.fieldName === 'unitCost')}
+                                    onChange={onValueChange}
+                            />
+                        </Col>
+                    }
+                    <Col xs={1}>
+                        <UnitQuantityInput
+                            value={model.quantity}
+                            hasError={model.errors?.find(field => field.fieldName === 'quantity')}
+                            onChange={onValueChange}
+                            isReadOnly={!isAdmin && canUpdate}
+                        />
+                    </Col>
+                    { (isAdmin || !canUpdate) &&
+                        <Col xs={1}>
+                            <SubtotalInput unitCost={model.unitCost} quantity={model.quantity} />
+                        </Col>
+                    }
+                    <Col xs={!hasRequestId ? 2 : 1}>
+                        <ControlledSubstance
+                            value={model.controlledSubstance}
+                            onChange={onValueChange}
+                            isReadOnly={!isAdmin && canUpdate}/>
+                    </Col>
+
+                    { hasRequestId && (isAdmin || canUpdate) &&
+                        <Col xs={1}>
+                            <QuantityReceivedInput
+                                value={model.quantityReceived}
+                                onChange={onValueChange}
+                            />
+                        </Col>
+                    }
+
+                    <Col xs={2}/>
+                    <Col xs={1}/>
```

Comment:
I'm not sure exactly what the layout of the page is, but could this space be given to any other inputs?

Reply:
1 is sufficient, otherwise all the inputs are too spaced out